### DISCUSSION TITLE
HAproxy as upstream server

### DISCUSSION THREAD

**Discussion by ErenYeager80** [mitmproxy/mitmproxy#5163] [General]
I used HAproxy server as an upstream server that load-balance between some proxy servers, but when I use HAproxy stickiness, no cookies are set...
How should I fix my problem?

**Comment by devcatkit** [comment]
hello

you maybe figure it out? tt has to do with this sticky-cookie thing but it seems not work 

https://docs.mitmproxy.org/stable/overview-features/#sticky-cookies

**Reply by chribro88** [reply]
I'm not super savvy the terminology, but I think (at least with HTTPS requests) HAProxy is setting the cookie on the CONNECT/ Proxy header/layer, whereas the you want it set on the HTTP request/response layer.

You either need to make HAProxy a transparent forward proxy, so essentially a MITM, or find a way for mitmproxy to pass the cookie between the layers.

That's my very limited understanding 😅 (I'm also trying to achieve similar with HAProxy but stick on a header value)
